Entities/AHuman.cpp

```diff
@@ -3565,51 +3605,39 @@ void AHuman::Update()
-                                                                     Matrix(),
-                                                                     m_Paths[FGROUND][WALK],
-                                                                     deltaTime,
-                                                                     &playStride,
-                                                                     false);
-            }
-            else
-                m_ArmClimbing[BGROUND] = false;
-
-            if (m_pBGLeg && (!m_pFGLeg || !(m_Paths[BGROUND][WALK].PathEnded() && FGLegProg < 0.5)))
-            {
-                m_StrideStart = false;
-                // Reset the stride timer if the path is about to restart
-                if (m_Paths[BGROUND][WALK].PathEnded() || m_Paths[BGROUND][WALK].PathIsAtStart())
-                    m_StrideTimer.Reset();
-                m_ArmClimbing[FGROUND] = !m_pBGFootGroup->PushAsLimb(m_Pos +
-                                                                     m_pBGLeg->GetParentOffset().GetXFlipped(m_HFlipped),
-                                                                     m_Vel,
-                                                                     Matrix(),
-                                                                     m_Paths[BGROUND][WALK],
-                                                                     deltaTime,
-                                                                     &playStride,
-                                                                     false);
-            }
-            else
-                m_ArmClimbing[FGROUND] = false;
+			if (m_pFGLeg && (!m_pBGLeg || !(m_Paths[FGROUND][WALK].PathEnded() && BGLegProg < 0.5F) || m_StrideStart)) {
+				// Reset the stride timer if the path is about to restart.
+				if (m_Paths[FGROUND][WALK].PathEnded() || m_Paths[FGROUND][WALK].PathIsAtStart()) { m_StrideTimer.Reset(); }
+				m_ArmClimbing[BGROUND] = !(m_pFGFootGroup->PushAsLimb(m_Pos + RotateOffset(m_pFGLeg->GetParentOffset()), m_Vel, m_WalkAngle[FGROUND], m_Paths[FGROUND][WALK], deltaTime, &restarted, false));
+				if (restarted) { UpdateWalkAngle(FGROUND); }
+			} else {
+				m_ArmClimbing[BGROUND] = false;
+			}
+			if (m_pBGLeg && (!m_pFGLeg || !(m_Paths[BGROUND][WALK].PathEnded() && FGLegProg < 0.5F))) {
+				m_StrideStart = false;
+				// Reset the stride timer if the path is about to restart.
+				if (m_Paths[BGROUND][WALK].PathEnded() || m_Paths[BGROUND][WALK].PathIsAtStart()) { m_StrideTimer.Reset(); }
+				m_ArmClimbing[FGROUND] = !m_pBGFootGroup->PushAsLimb(m_Pos + RotateOffset(m_pBGLeg->GetParentOffset()), m_Vel, m_WalkAngle[BGROUND], m_Paths[BGROUND][WALK], deltaTime, &restarted, false);
+				if (restarted) { UpdateWalkAngle(BGROUND); }
+			} else {
+				if (m_pBGLeg) { m_pBGFootGroup->FlailAsLimb(m_Pos, RotateOffset(m_pBGLeg->GetParentOffset()), m_pBGLeg->GetMaxLength(), m_PrevVel, m_AngularVel, m_pBGLeg->GetMass(), deltaTime); }
```

Comment:
The BG leg is set to flail according to `m_Vel` for the duration of the first step because otherwise it will linger at the previous spot and look dumb

Basically this makes the leg follow the body better for the very first step